### PR TITLE
Homework 3 (done)

### PR DESCRIPTION
**Pull request by ilkivs**
homework 3 pull request

### CODE REVIEW

File: src/HomeWork/hw3/task1/MergeSort.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+    }
+
+    private void initArrays() {
+        Random r = new Random();
+        int firstArrSize = r.nextInt(10);
+        int secondArrSize = r.nextInt(10);
+        firstArray = new int[firstArrSize];
+        secondArray = new int[secondArrSize];
+        resultArray = new int[firstArray.length + secondArray.length];
+        for (int i = 0; i < firstArray.length; i++) {
```
Comment (Siril-O) [inline src/HomeWork/hw3/task1/MergeSort.java]: move this to method it is duplicated

File: src/HomeWork/hw3/task1/MergeSort.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+            firstArray[i] = r.nextInt(100);
+        }
+        for (int i = 0; i < secondArray.length; i++) {
+            secondArray[i] = r.nextInt(100);
```
Comment (Siril-O) [inline src/HomeWork/hw3/task1/MergeSort.java]: make 100 a constant

File: src/HomeWork/hw3/task1/MergeSort.java [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+        }
+        Arrays.sort(firstArray);
```
Comment (Siril-O) [inline src/HomeWork/hw3/task1/MergeSort.java:85]: Why do you sort them ?

File: src/HomeWork/hw3/task1/MergeSort.java (hunk 1 above)
Comment (Siril-O) [reply src/HomeWork/hw3/task1/MergeSort.java]: create method which creates random array and call it to times  when you init your firstArray  secondArray fields

**Review by Siril-O — CHANGES_REQUESTED** [review]

File: src/HomeWork/hw3/task1/MergeSort.java (hunk 3 above)
Comment (ilkivs) [reply src/HomeWork/hw3/task1/MergeSort.java:85]: because input arrays must be sorted before merge sort

Comment (Siril-O) [reply src/HomeWork/hw3/task1/MergeSort.java:85]: this is merge of arrays not merge sort, merge sort should sort arrays itself

**Review by Siril-O — APPROVED** [review]

File: src/HomeWork/hw3/task1/MergeSort.java (hunk 3 above)
Comment (ilkivs) [reply src/HomeWork/hw3/task1/MergeSort.java:85]: ok, thank you :)

**Pull request merged**